### PR TITLE
[deps] Bump sdk-registry and access-token gradle plugin versions.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Enable gradle rich version by bumping sdk-registry gradle plugin version.</changelog>`.

### Summary of changes

This PR bumps the sdk-registry gradle plugin version to 0.7.0 and access-token gradle plugin version to 0.4.0.

Sdk-registry gradle plugin 0.7.0 introduced the support for uploading the [Gradle Module Metadata](https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html) file and enables better dependency management.

### CODE REVIEW

File: android-auto-app/build.gradle.kts [hunk 1]
```diff
@@ -17,8 +17,10 @@ android {
     versionName = "0.1.0"
     multiDexEnabled = true
     testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
-    testInstrumentationRunnerArguments = mapOf(
-      "clearPackageData" to "true"
+    testInstrumentationRunnerArguments(
+      mapOf(
```
Comment (alexander-kulikovskii) [inline android-auto-app/build.gradle.kts:21]: does it means that you should bump GP version?

Comment (pengdev) [reply android-auto-app/build.gradle.kts:21]: Still trying to figure out why we need to change these, seems the new gradle plugin depends on AGP 4.1 and the current AGP version in the mapbox-maps-android is 4.0.

Probably we should bump the AGP version in this repo as well.

Comment (alexander-kulikovskii) [reply android-auto-app/build.gradle.kts:21]: it will be nice :slightly_smiling_face:
mutation tests will be merged

**Review by Chaoba — APPROVED** [review]

File: android-auto-app/build.gradle.kts (hunk 1 above)
Comment (pengdev) [reply android-auto-app/build.gradle.kts:22]: We will need to bump AGP version from upstream first, for now let's keep it as is.

**Review by kiryldz — APPROVED** [review]
🧹

**Comment by kiryldz** [comment]
@pengdev the only thing to mention: are benchmark CI job failure connected with your changes here?

**Comment by pengdev** [comment]
@kiryldz yes, and I created a PR to fix it and linked to this ticket.

**Comment by pengdev** [comment]
Closing in favour of https://github.com/mapbox/mapbox-maps-android/pull/1118

**Pull request closed without merging**